src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyState = { ...state };
+  const massiveToReturn = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(copyState, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete copyState[key];
+      }
+    }
+
+    if (action.type === 'clear') {
+      for (const prop in copyState) {
+        delete copyState[prop];
+      }
+    }
+    massiveToReturn.push({ ...copyState });
```

Comment:
[DONT REPEAT YOURSELF] - Pushing the copy of the object to the array should be done at the end of the loop cycle but outside of the `switch` block, after the current action is processed. This avoids repetition of the push operation in every case.